packages/eas-cli/src/submissions/types.ts

```diff
@@ -1,55 +1,18 @@
-export interface BaseSubmissionContext {
+import { AndroidSubmitProfile, IosSubmitProfile } from '@expo/eas-json';
+
+import { AppPlatform } from '../graphql/generated';
+
+export interface SubmissionContext<T extends AppPlatform> {
+  archiveFlags: SubmitArchiveFlags;
+  platform: T;
+  profile: T extends AppPlatform.Android ? AndroidSubmitProfile : IosSubmitProfile;
   projectDir: string;
   projectId: string;
 }
```

Comment:
```suggestion
export interface SubmissionContext<T extends Platform> {
  archiveFlags: SubmitArchiveFlags;
  platform: T;
  profile: SubmitProfile<T>;
  projectDir: string;
  projectId: string;
}
```
maybe you could use Platform type and remap to AppPlatform when you need it?